src/shims/unix/unnamed_socket.rs

```diff
@@ -237,6 +241,7 @@ impl FileDescription for AnonSocket {
         if let Some(clock) = &ecx.release_clock() {
             writebuf.clock.join(clock);
         }
+        let _ = ecx.release_clock().is_some(); // Increment clock again, to establish a happens-after timeframe
```

Comment:
Something seems quite wrong if you create a clock just to throw it away then.

And I still haven't understood what problem you are even trying to fix. Do you have a testcase demonstrating the issue?

Reply:
That's a good question to resolve first. What's the other problem I'm trying to fix? I've created #3947 for that.

Reply:
> Do you have a testcase demonstrating the issue?

Yes, `test_issue_3944_with_just_one_event`. But what it demonstrated before and then after the fix is now secondary to the discussion I hope we can have in #3947.